lib/signatures/signature-helper.ts

```diff
@@ -0,0 +1,53 @@
+import { createHmac, timingSafeEqual } from "node:crypto";
+import { parseWebhookResponse } from "../models/parse-webhook.js";
+import type { WebhookResponse } from "../models/webhook-schemas.js";
+import type { ParseResult } from "../utils/types.js";
+
+export const parseSignedWebhookResponse = (
+  jsonPayload: string,
+  secret: string,
+  signature: string,
+): ParseResult<WebhookResponse> => {
+  if (!isSignatureValid(jsonPayload, secret, signature)) {
+    return { success: false, error: new Error("Webhook signature validation failed") };
+  }
+
+  try {
+    const parsedPayload = JSON.parse(jsonPayload) as unknown;
+
+    return parseWebhookResponse(parsedPayload);
+  } catch (error) {
+    return {
+      success: false,
+      error: new Error(
+        `Failed to parse webhook payload: ${error instanceof Error ? error.message : "Invalid JSON"}`,
+      ),
+    };
+  }
+};
+
+export const isSignatureValid = (
```

Comment:
Might be personal preference, but I like to use destructured object params instead of (string, string, string) as it's easier to read and understand at a glance. Passing 3 strings next to each other might obscur their meaning unless everything is named nicely:)